src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +17,34 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+        } catch (IllegalArgumentException e) {
+            throw new DataProcessingException("The instance of Country is not an entity.", e);
+        } catch (IllegalStateException | PersistenceException e) {
```

Comment:
Consider refining the exceptions used here. `IllegalArgumentException` might not be the most appropriate for non-entity instances, and `IllegalStateException` is not typically related to persistence operations. It might be beneficial to review these choices.